### PR TITLE
translate stransform.py into js

### PR DESCRIPTION
**Pull request by jihan1218**
Check whether the translation is correct

### CODE REVIEW

File: pysc2/pysc2/lib/transform.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
+
+const path = require('path')
+const point = require(path.resolve(__dirname, './point.js'))
+
+class Transform extends object {
+	fwd_dist(dist){
+		throw NotImplementedError()
```
Comment (arcman7) [inline pysc2/pysc2/lib/transform.js:7]: where do these classes come from, NotImplementedError?

Comment (jihan1218) [reply pysc2/pysc2/lib/transform.js:7]: It looks like it is from python built in exceptions. Should we implement it in js?

File: pysc2/pysc2/lib/pythonUtils.js [hunk 2]
```diff
@@ -156,3 +156,30 @@ module.exports = {
   withPython,
   zip,
 }
+
```
Comment (arcman7) [inline pysc2/pysc2/lib/pythonUtils.js:159]: Take note of modules.export

It alone holds a reference to everything that is created and exported from this file. Your function definition should go above it, and then add your function to the exports object

File: pysc2/pysc2/lib/transform.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
+
+const path = require('path')
+const point = require(path.resolve(__dirname, './point.js'))
+const pythonUtils = require(path.resolve(__dirname, './pythonUtils.js'))
+
+NotImplementedError = pythonUtils.NotImplementedError
```
Comment (arcman7) [inline pysc2/pysc2/lib/transform.js:6]: const NotImplementedError = pythonUtils.NotImplementedError

File: pysc2/pysc2/lib/transform.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    super(scale, offset)
+    if (scale == null) {
+      this.scale = new point.Point(1, 1)
+    } else if (isinstance(scale, Number)) {
+      this.scale = new point.Point(scale, scale)
+    } else {
+      this.scale = scale
+    }
+    assert(this.scale.x !== 0 && this.scale.y !== 0)
+    this.offset = offset || point.Point(0, 0)
```
Comment (arcman7) [inline pysc2/pysc2/lib/transform.js:36]: this.offset = offset || new point.Point(0, 0)

**Pull request merged**